packages/account/src/providers/transaction-request/transaction-request.ts

```diff
@@ -249,6 +265,32 @@ export abstract class BaseTransactionRequest implements BaseTransactionRequestLi
     this.witnesses[index] = witness;
   }
 
+  /**
+   * Helper function to add an external signer to the transaction.
+   *
+   * @param signer - The signer to add to the transaction.
+   * @returns The transaction with the signature witness added.
+   */
+  async addSigner(signer: Account) {
+    const hashedTransaction = this.getTransactionId(signer.provider.getChainId());
+    const signature = await signer.signTransaction(hashedTransaction);
+    this.addWitness(signature);
+
+    return this;
+  }
+
+  /**
+   * Adds multiple external signers to the transaction.
+   *
+   * @param signers - The signers to add to the transaction.
+   * @returns The transaction with multiple signature witnesses added.
+   */
+  async addSigners(signers: Account[]) {
```

Comment:
IMO there is no need to have these plural-style methods because whoever wants to add multiple signers will just create their own ad-hoc implementation just like the one in this method:

```ts
await Promise.all(signers.map((signer) => request.addSigner(signer)));
```

It adds an additional maintenance burden on us because we have to keep the interfaces _and_ descriptions of both methods in sync. I already had to do this one time in another place and it's generally a PITA because you always forget something.

Reply:
Hmm I think I like the pluralised as the use case for a multi sig is pretty common, `addSigners` will probably be more common place than `addSigner`. I can appreciate the additional burden. What about we ditch `addSigner`, only keep `addSigners` and I'll move all the logic there?

Reply:
I'm okay with both, but if we make signing synchronous as per the implications of https://github.com/FuelLabs/fuels-ts/pull/1815#discussion_r1511520861, it would be as straightforward as this then:
```ts
signers.forEach(signer => request.addSigner(signer))
```

I like this more than having the plural because I don't like the one-signer use case where an array has to be instantiated just for the signing to work: 
```ts
request.addSigners([signer]);
```

This is just my preference, so if you like the `addSigners` more, I'm okay with you implementing that as well.

Reply:
> I'm okay with both, but if we make signing synchronous as per the implications of https://github.com/FuelLabs/fuels-ts/pull/1815#discussion_r1511520861, it would be as straightforward as this then:

I shall make the signing synchronous, good spot 👍🏻 

> I like this more than having the plural

My preference here is I do like with the explicit `addSigners` as it can be chained onto the tx request for better DX

> This is just my preference, so if you like the addSigners more, I'm okay with you implementing that as well.

True and likewise, I'm appreciating the discussion though 👍🏻  May see if anyone else chimes in with a preference.

Reply:
Ah, good one with the chaining! I'm on your side now :smile:

Reply:
I have now implemented `addAccountWitnesses` which takes `Account | Account[]`.